src/modules/app.js

```diff
@@ -0,0 +1,23 @@
+import getData from './getData.js';
+import popup from './commentPopup.js';
+
+const button = document.querySelector('button');
+
+button.addEventListener('click', async (e) => {
+  let appData = [];
+  try {
+    appData = await getData();
+  } catch (e) {
+    return `Error ${e}`;
+  }
+  const data = appData.filter((item) => item.id === +(e.target.id));
```

Comment:
It would be better if you use then to wait for the async data
 `getData().then((response)=>{
})`
Refer to [await function mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)